llvm/lib/Target/AArch64/AArch64TargetMachine.cpp

```diff
@@ -472,7 +482,8 @@ AArch64TargetMachine::getSubtargetImpl(const Function &F) const {
     resetTargetOptions(F);
     I = std::make_unique<AArch64Subtarget>(
         TargetTriple, CPU, TuneCPU, FS, *this, isLittle, MinSVEVectorSize,
-        MaxSVEVectorSize, IsStreaming, IsStreamingCompatible, HasMinSize);
+        MaxSVEVectorSize, IsStreaming, IsStreamingCompatible,
+        StreamingHazardSize, HasMinSize);
```

Comment:
I never know this layering very well, but is there a strong reason to go via AArch64TargetMachine, as opposed to just setting the variable in AArch64Subtarget::AArch64Subtarget? I would expect this might become a target-feature in the end, which could be overridden by the option. That might be simpler if it was contained within AArch64SubTarget.

Reply:
Not particularly, I was mainly following how other values are passed via the constructor. Are you suggesting to move the `cl::opt` to the top of `AArch64Subtarget.cpp` instead (then set the member value in the ctor body)?

Reply:
The streaming and vector size properties come from function attributes, so they need to be passed to the subtarget when constructing it in `AArch64TargetMachine::getSubtargetImpl(const Function &)`.
For the option in this PR there is no corresponding function attribute and therefore no need for the indirection, so it makes sense to move the option to AArch64Subtarget.cpp.

Reply:
Ah that makes sense, moved the option now :+1: